cleanupNoncurrentVersions.js

```diff
@@ -327,18 +343,27 @@ function _triggerDeletesOnEligibleObjects(bucket, versions, deleteMarkers,
                    deleteMarkers[matchingDeleteMarkerIndex].Key < version.Key) {
                 ++matchingDeleteMarkerIndex;
             }
-            if (matchingDeleteMarkerIndex < deleteMarkers.length &&
-                deleteMarkers[matchingDeleteMarkerIndex].Key === version.Key &&
-                deleteMarkers[matchingDeleteMarkerIndex].IsLatest === true) {
+            const isDeleted = matchingDeleteMarkerIndex < deleteMarkers.length
+                && deleteMarkers[matchingDeleteMarkerIndex].Key === version.Key
+                && deleteMarkers[matchingDeleteMarkerIndex].IsLatest === true;
+
+            const isElligible = _deleteMarkerIsEligible(
+                deleteMarkers[matchingDeleteMarkerIndex].LastModified);
+
+            if (isDeleted && isElligible) {
```

Comment:
This should be conditioned to the existence of the delete marker:

```suggestion
            const isEligible = isDeleted && _deleteMarkerIsEligible(
                deleteMarkers[matchingDeleteMarkerIndex].LastModified);

            if (isEligible) {
```